### PR TITLE
[MFE-HS] Template mapping, working table widget

### PR DESCRIPTION
**Pull request by rimselis**
Added automated template mapping (refer to README)
Updated table to at least display data

### CODE REVIEW

File: generators/entity-server/scripts/mfe-files.js.ejs [hunk 1]
```diff
@@ -0,0 +1,32 @@
+const files = {
+    path: 'ui/widgets',
+    templates: [
+        <%_ files.map(file => { _%>
+        <%_ if (file.options.skip) { _%>
+        /*
+        <%_ } _%>
```
Comment (joewhite101) [inline generators/entity-server/scripts/mfe-files.js.ejs]: Do we need the commented out section or can we remove it?

File: generators/entity-server/scripts/createMfeTemplateMap.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
+const fs = require('fs');
+const path = require('path');
+const ejs = require('ejs');
+
+const TEMPLATES_PATH = '../templates/ui/';
+const EXPORT_PATH = '../mfe-files.js';
+
+const IGNORED_TEMPLATES = [
+    // `${TEMPLATES_PATH}widgets/common/jsconfig.json.ejs`,
```
Comment (joewhite101) [inline generators/entity-server/scripts/createMfeTemplateMap.js]: Remove if dead

File: generators/entity-server/files.js [hunk 3]
```diff
@@ -30,10 +27,22 @@ const serverFiles = {
 
 const microFrontEndFiles = {
     server: [
-        // eslint-disable prettier/prettier
-        detailsWidgetFiles.dynamic,
-        detailsWidgetFiles.copy,
-        tableWidgetFiles,
+        widgetFiles,
+        // {
```
Comment (joewhite101) [inline generators/entity-server/files.js]: Is this dead?

File: generators/entity-server/scripts/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    },
+  ],
+};
+
+module.exports = files;
+```
+
+Add `<%# { "useBluePrint": true } -%>` to the beginning of each template that is dynamic (not just copied) and that will add the content of JSON object to the mapping (in this case `{ useBluePrint: true }` would be added).
+
+The script keeps the file structure and parses all the files. If you want a file to be skipped (usually this should not happen), add a `<%# { "skip": true } -%>` option. To add custom mappings (e.g., to change the file name), just add `renameTo` option: ``{ "renameTo": "`/${generator.entityInstance}/tableWidget/src/mocks/${generator.entityInstance}i18nMock.js`" }``.
```
Comment (joewhite101) [inline generators/entity-server/scripts/README.md]: Very cool

File: generators/entity-server/scripts/mfe-files.js.ejs (hunk 1 above)
Comment (rimselis) [reply generators/entity-server/scripts/mfe-files.js.ejs]: My approach to this is to have the "skipped" files still visible somewhere for us to know they exist. Currently, tests are skipped as I did not found a good way to write those inside the .ejs.
In the end, there should be no "skipped" files in the file structure and these would not be in the generated file

File: generators/entity-server/files.js (hunk 3 above)
Comment (rimselis) [reply generators/entity-server/files.js]: This was for shared resources, but I removed those. Will remove

**Pull request merged**